### DISCUSSION TITLE
do not understand the inference process of xtts 2.0

### DISCUSSION THREAD

**Discussion by Liujingxiu23** [coqui-ai/TTS#3468] [General Q&A]
I understand the process is:
1.get "cond_latents" from the prompt waves and get text_embeding from the input text
2.gpt_codes = gpt_inference.generate(text, cond_latents)
3.gpt_latents = gpt(text, gpt_codes,cond_latents)
4.HifiDecoder

why 2+3?

**Comment by Approximetal** [comment]
https://huggingface.co/blog/how-to-generate
It might help to explain step 2
